launcher/src/se/llbit/chunky/launcher/JavaFxLocator.java

```diff
@@ -0,0 +1,82 @@
+package se.llbit.chunky.launcher;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class JavaFxLocator {
+
+
+  private static final String[] javafxPathCandidates;
+
+  static {
+    String[] javafxPathCandidates1;
+    try {
+      javafxPathCandidates1 = new String[]{
+              "C:/Program Files/openjfx/lib",
+              System.getProperty("java.home", "") + File.separator + "lib", // java home
+              new File(ChunkyLauncher.class.getProtectionDomain().getCodeSource().getLocation().toURI())
+                      .toPath().getParent().toAbsolutePath().toString() + File.separator + "lib", // directory of the .jar
+              System.getProperty("user.dir", "") + File.separator + "lib", // working directory
+              "/usr/share/openjfx/lib",
+              "/usr/lib/jvm/java-11-openjdk/lib",
+              "/usr/lib/jvm/java-12-openjdk/lib",
+              "/usr/lib/jvm/java-13-openjdk/lib",
+              "/usr/lib/jvm/java-14-openjdk/lib",
+              "/usr/lib/jvm/java-15-openjdk/lib",
+      };
+    } catch(URISyntaxException e) {
+      e.printStackTrace();
+      javafxPathCandidates1 = new String[0];
+    }
+    javafxPathCandidates = javafxPathCandidates1;
+  }
+
+  private static boolean validJavafxDirectory(Path dir) {
+    return dir.toFile().exists()
+            && dir.resolve("javafx.base.jar").toFile().exists()
+            && dir.resolve("javafx.controls.jar").toFile().exists()
+            && dir.resolve("javafx.graphics.jar").toFile().exists()
+            && dir.resolve("javafx.fxml.jar").toFile().exists();
+  }
+
+  private static void runWithJavafx(Path javafxDir, String[] args) {
+    // https://stackoverflow.com/questions/4159802/how-can-i-restart-a-java-application
+    ArrayList<String> cmd = new ArrayList<>();
+    cmd.add(System.getProperty("java.home") + File.separator + "bin" + File.separator + "java");
+    cmd.addAll(ManagementFactory.getRuntimeMXBean().getInputArguments());
+    cmd.add("--module-path");
+    cmd.add(javafxDir.toAbsolutePath().toString());
+    cmd.add("--add-modules");
+    cmd.add("javafx.controls,javafx.fxml");
+
+    cmd.add("-cp");
+    cmd.add(ManagementFactory.getRuntimeMXBean().getClassPath());
+    cmd.add(ChunkyLauncher.class.getName());
+    cmd.addAll(Arrays.asList(args));
+    cmd.add("--noRetryJavafx"); // Make sure this doesn't end up as a fork bomb
+
+    ProcessBuilder builder = new ProcessBuilder(cmd);
+    builder.inheritIO();
+    try {
+      Process process = builder.start();
+      System.exit(process.waitFor());
```

Comment:
Why wait for the subprocess?

Reply:
To give the illusion that there is only one process and because we want the output/error stream to be linked with the child streams (I don't know if the stream redirection still works if exiting immediatly)